portfolio/src/components/layout.tsx

```diff
@@ -15,24 +14,25 @@ const Page = Styled.div`
     color: ${theme.primary};
     text-decoration: none;
   }
-`
+`;
 
 const PageBody = Styled.div`
-  width: 100%;
-  max-width: 1191px;
+  width: 800px;
+  max-width: 100%;
   margin-top: 60px;
-`
-
+`;
 
-const Layout = ({ children }: { children: Array<React.ReactChild> | React.ReactChild }) => {
+const Layout = ({
+  children,
+}: {
+  children: Array<React.ReactChild> | React.ReactChild;

```

Comment:
? not sure what this is doing

Reply:
Types are declared when you create a variable (or argument of a function like seen here)

examples:
`const name: string = "Gage"`;
`const age: number = 28`;

You can set types inline, like this:
`const weather: { zip: string, temperature: number } = { zip: "98121", temperature: 71 }`

or you can create a type as a variable by using the keyword `type` instead of `const`
```ts

type Weather = {
  zip: string,
  temperature: string
}

const WeatherComponent = ( { zip, temperature } : Weather) => { ... }
```

You set the type so that code using these variables will only allow those types of variables, preventing you from coding incorrectly by using variables the wrong way. ie. It doesn't allow you to use a number where a string is required. It doesn't allow you to access `weather.humidity` because the property `humidity` isn't set in the type.

This is a type for the `children` props on the component. React gives us the types to use - types could be `string`, `bool`, `number`, etc. but there are also more complicated types that are objects or functions such as `React.ReactChild`, `WeatherItems`, etc. 

```html
<Layout>
   # anything in here goes into the `children` prop on a component (this is a react thing)
</Layout
```

example:
```html
<Layout>
   <h1>My name is Gage</h1>
</Layout
```
```js
const Layout = ({children}) => {
  // children = <h1>My name is Gage</h1> 
  // because children is just 1 element, the type would be React.ReactChild

 return <div>{children}</div> // returns <div><h1>My name is Gage</h1></div>
}
```

children can either be 1 element or an array of elements
The type here would be: Array<React.ReactChild>
example of `children` being an array:
```
<Layout>
   <h1>My name is Gage</h1>
   <p>I live in Seattle WA and work as a software engineer</p>
</Layout>
```

Array<type> is how you declare a type of multiples

`Array<string>`
`Array<{ name: string }`

`Array<number>`
`Array<React.ReactChild>`

Because I wanted the component `Layout` to allow a single element or an array of elements, i set the prop as an "or" which is done by adding a pipe `|`

`children: React.ReactChild | Array<React.ReactChild> ` 

https://www.typescriptlang.org/docs/handbook/typescript-from-scratch.html is a good place to start for typescript.